boa/network.py

```diff
@@ -419,6 +419,8 @@ def _send_txn(self, from_, to=None, gas=None, value=None, data=None):
         print(f"tx broadcasted: {tx_hash}")
 
         receipt = self._rpc.wait_for_tx_receipt(tx_hash, self.tx_settings.poll_timeout)
+        if receipt["status"] != "0x1":
+            raise RuntimeError(f"txn failed: {receipt}")
```

Comment:
why RuntimeError?

Reply:
BoaError expects a stack trace. Which should we use?

Reply:
RuntimeError might be ok, but i'm not sure. i think the only way we can reach here is if the fork mode simulation did not raise any errors, but there is an error when actually sending the transaction (presumably caused by a state change between our snapshot and now).

Reply:
This was the example in the issue https://bscscan.com/tx/0x77967045c0b62986ad3cf8933c746dff0b448dbffb2e71b89869e64ff54d1012

Reply:
hmm, i'm a bit confused why it's not already being in one of these two places:

https://github.com/vyperlang/titanoboa/blob/98c7b5884536cc5c333424a2d257b0443eb1c319/boa/network.py#L251
or
https://github.com/vyperlang/titanoboa/blob/98c7b5884536cc5c333424a2d257b0443eb1c319/boa/network.py#L316